pkg/api/componentreadiness/component_report.go

```diff
@@ -608,6 +677,180 @@ func (s *sampleQueryGenerator) queryTestStatus() (crtype.ReportTestStatus, []err
 	return crtype.ReportTestStatus{SampleStatus: sampleStatus}, errs
 }
 
+type fallbackTestQueryReleasesGenerator struct {
+	client                     *bqcachedclient.Client
+	cacheOption                cache.RequestOptions
+	commonQuery                string
+	groupByQuery               string
+	queryParameters            []bigquery.QueryParameter
+	allJobVariants             crtype.JobVariants
+	BaseRelease                string
+	BaseStart                  time.Time
+	BaseEnd                    time.Time
+	CachedFallbackTestStatuses crtype.FallbackReleases
+}
+
+func (f *fallbackTestQueryReleasesGenerator) getTestFallbackReleases() (*crtype.FallbackReleases, []error) {
+	wg := sync.WaitGroup{}
+	fallbackRelease := f.BaseRelease
+	f.CachedFallbackTestStatuses = newFallbackReleases()
+
+	releases, errs := GetReleaseDatesFromBigQuery(f.client)
+
+	if errs != nil {
+		return nil, errs
+	}
+
+	var err error
+	// currently gets current base plus previous 3
+	// current base is just for testing but use could be
+	// extended to no longer require the base query
+	for i := 0; i < 4; i++ {
+
+		// currently fetching the baseRelease via fallback to
+		// perform validation of the test data
+		// this gets uncommented when ready
+		// and the section at the bottom removed
+		// fallbackRelease, err = previousRelease(fallbackRelease)
+		// if err != nil {
+		// 	log.WithError(err).Errorf("Failure determining fallback release for %s", fallbackRelease)
+		// 	continue
+		// }
+
+		start := f.BaseStart
+		end := f.BaseEnd
+		var crRelease *crtype.Release
+
+		for _, release := range releases {
+			if release.Release == fallbackRelease {
+				crRelease = &release
```

Comment:
```suggestion
		for i, release := range releases {
			if release.Release == fallbackRelease {
				crRelease = &releases[i]
```

Or [update go.mod to require 1.22](https://go.dev/blog/loopvar-preview)